tests/PHPStan/Analyser/TypeSpecifierTest.php

```diff
@@ -422,7 +422,7 @@ public function dataCondition(): array
 					new Variable('foo'),
 					new Variable('bar'),
 				),
-				['$foo' => 'Bar', '$bar' => 'Bar'],
+				['$foo' => 'Bar', '$bar' => 'mixed'], // could be '$bar' => 'Bar'
```

Comment:
This is really annoying to not have anymore :/

I might spend some time on this in the future to bring back.

Reply:
I'm sorry, it looked pretty esoteric to me as a way to narrow down types :)

Reply:
Maybe we can still do it only for objects, or something like that...

Reply:
If it annoys me enough I will find a way to bring it back without a huge performance loss :)